posthog/temporal/batch_exports/utils.py

```diff
@@ -24,3 +29,33 @@ def rewind_gen() -> collections.abc.Generator[T, None, None]:
             yield i
 
     return (first, rewind_gen())
+
+
+async def try_set_batch_export_run_to_running(run_id: str | None, logger, timeout: float = 10.0) -> None:
+    """Try to set a batch export run to 'RUNNING' status, but do nothing if we fail or if 'run_id' is 'None'.
+
+    This is intended to be used within a batch export's 'insert_*' activity. These activities cannot afford
+    to fail if our database is experiencing issues, as we should strive to not let issues in our infrastructure
+    propagate to users. So, we do a best effort update and swallow the exception if we fail.
+
+    Even if we fail to update the status here, the 'finish_batch_export_run' activity at the end of each batch
+    export will retry indefinitely and wait for postgres to recover, eventually making a final update with
+    the status. This means that, worse case, the batch export status won't be displayed as 'RUNNING' while running.
+    """
+    if run_id is None:
+        return
+
+    try:
+        await asyncio.wait_for(
+            asyncio.to_thread(
+                update_batch_export_run,
+                uuid.UUID(run_id),
+                status=BatchExportRun.Status.RUNNING,
+            ),
+            timeout=timeout,
+        )
+    except Exception as e:
+        logger.warn(
+            "Unexpected error trying to set batch export to 'RUNNING' status. Run will continue but displayed status may not be accurate until run finishes",
+            exc_info=e,
+        )
```

Comment:
TODO: This log line should probably not be shown to users. Right now we don't have a way to separate logs to users and internal logs for debugging. We need a better error handling + failure notification + log solution. I'm cooking some stuff but it will take a while :cook: